### DISCUSSION TITLE
The number of quantile bins we have to choose

### DISCUSSION THREAD

**Discussion by muhshif** [btschwertfeger/python-cmethods#27] [Q&A]
In Quantile mapping methods (Quantile mapping & Quantile delta mapping), how much quantile do we have to choose actually? 1000? or 100? or anything else? What is it actually dependent on? Why?

**Comment by btschwertfeger** [comment]
Selecting the appropriate number of quantiles is a challenging task. The number of quantiles determines the number of bins in the probability density function and the cumulative distribution function, which represent the distribution of the time series. The number of quantiles should be large enough to capture the nuances of the distribution functions but not excessively high, as this can result in a straight-line distribution.

In my bachelor's thesis, I recall using 100 quantiles to adjust 30 years of daily data (10950 values). In my latest article (https://www.sciencedirect.com/science/article/pii/S2352711023000754), I employed 1000 quantiles for the same period. The choice of quantile value is often determined through trial and error.

Calculating the quantiles using python-cmethods can be time-consuming, particularly when using many iterations and a high number of quantiles. However, with [BiasAdjustCXX](https://github.com/btschwertfeger/BiasAdjustCXX), the process is over a hundred times faster and memory usage is not a concern.
